### PR TITLE
che role

### PR DESCRIPTION
**Pull request by odra**
Adds che role installation role.

Verification:

```
#install
ansible-playbook -i roles/che/tests/inventory-multi-external-keycloak roles/che/tests/test-install.yml 

#uninstall
ansible-playbook -i roles/che/tests/inventory-multi-external-keycloak roles/che/tests/test-uninstall.yml 
```

ping @david-martin @maleck13 @aidenkeating

### CODE REVIEW

**Comment by david-martin** [comment]
@odra hitting an error pretty quick when running above cmd.
Will debug some

```TASK [che : Deploy postgres] ************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "oc new-app -p 'CHE_VERSION=6.9.0' -p 'IMAGE_POSTGRES=docker.io/eclipse/che-postgres' -f /tmp/che-templates/multi/postgres-template.yaml -n che-multi", "delta": "0:00:00.140968", "end": "2018-08-07 21:03:00.762799", "msg": "non-zero return code", "rc": 1, "start": "2018-08-07 21:03:00.621831", "stderr": "error: unable to locate any template files with name \"/tmp/che-templates/multi/postgres-template.yaml\"\n\nThe 'oc new-app' command will match arguments to the following types:\n\n  1. Images tagged into image streams in the current project or the 'openshift' project\n     - if you don't specify a tag, we'll add ':latest'\n  2. Images in the Docker Hub, on remote registries, or on the local Docker engine\n  3. Templates in the current project or the 'openshift' project\n  4. Git repository URLs or local paths that point to Git repositories\n\n--allow-missing-images can be used to point to an image that does not exist yet.\n\nSee 'oc new-app -h' for examples.", "stderr_lines": ["error: unable to locate any template files with name \"/tmp/che-templates/multi/postgres-template.yaml\"", "", "The 'oc new-app' command will match arguments to the following types:", "", "  1. Images tagged into image streams in the current project or the 'openshift' project", "     - if you don't specify a tag, we'll add ':latest'", "  2. Images in the Docker Hub, on remote registries, or on the local Docker engine", "  3. Templates in the current project or the 'openshift' project", "  4. Git repository URLs or local paths that point to Git repositories", "", "--allow-missing-images can be used to point to an image that does not exist yet.", "", "See 'oc new-app -h' for examples."], "stdout": "", "stdout_lines": []}
```

File: evals/roles/che/templates/deploy/che-server-template.yaml [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  value: 'unique'
+- name: CHE_KEYCLOAK_ADMIN_REQUIRE_UPDATE_PASSWORD
+  displayName: Admin password update
+  description: Force an admin to update password after 1st login. True by default
+  value: 'true'
+- name: CHE_KEYCLOAK_REALM
+  displayName: Identity provider Realm
+  description: Identity provider Realm. Defaults to che
+  value: 'che'
+- name: CHE_KEYCLOAK_CLIENT_ID
```
Comment (david-martin) [inline evals/roles/che/templates/deploy/che-server-template.yaml]: @odra this should be `CHE_KEYCLOAK_CLIENT__ID` now, yes?

File: evals/roles/che/templates/deploy/multi/oauth-client.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2012-2018 Red Hat, Inc
```
Comment (david-martin) [inline evals/roles/che/templates/deploy/multi/oauth-client.yaml]: @odra this shouldn't be needed as che will be using openshift's keycloak

File: evals/roles/che/templates/deploy/https/keycloak-route-tls.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2012-2018 Red Hat, Inc
```
Comment (david-martin) [inline evals/roles/che/templates/deploy/https/keycloak-route-tls.yaml]: @odra this shouldn't be needed as che will be using openshift's keycloak

File: evals/roles/che/templates/deploy/multi/keycloak-template.yaml [hunk 4]
```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2012-2018 Red Hat, Inc
```
Comment (david-martin) [inline evals/roles/che/templates/deploy/multi/keycloak-template.yaml]: @odra this shouldn't be needed as che will be using openshift's keycloak

File: evals/roles/che/templates/deploy/che-server-template.yaml (hunk 1 above)
Comment (odra) [reply evals/roles/che/templates/deploy/che-server-template.yaml]: just the container env var, this is the template param env var

Comment (david-martin) [reply evals/roles/che/templates/deploy/che-server-template.yaml]: I changed both in the pr https://github.com/eclipse/che/pull/10646/files#diff-9f9225d85a063de310ccab9619413f7cR124

**Comment by david-martin** [comment]
@odra 1 small thing i noticed, which won't block the merge.
Maybe we need a card for the uninstall also removing the Keycloak client?

**Comment by odra** [comment]
Yeah make sense, I can work on it in a separate pr.

**Pull request merged**